firecrown/likelihood/gauss_family/statistic/cluster_number_counts.py

```diff
@@ -0,0 +1,277 @@
+    data_vector : list of float
+        A list with the number of clusters in each bin of redshift and Mproxy.
+        Set after `read` is called.
+    theory_vector
+        A list with the theoretical prediction of the number of clusters in each bin
+        of redshift and Mproxy. Set after `compute` is called.
+    sacc_inds : list of str
+        A list with the indices for the data vector that corresponds to the type of data
+        provided in the SACC file. Set after `read` is called.
+    """
+
+    def __init__(
+        self,
+        sacc_tracer,
+        sacc_data_type,
+        number_density_func,
+        systematics: Optional[List[Systematic]] = None,
+        mu_p0: Optional[float] = 1.0,
+        mu_p1: Optional[float] = 1.0,
+        mu_p2: Optional[float] = 1.0,
+        sigma_p0: Optional[float] = 1.0,
+        sigma_p1: Optional[float] = 1.0,
+        sigma_p2: Optional[float] = 1.0,
+    ):
+
+        super().__init__()
+
+        self.sacc_tracer = sacc_tracer
+        self.sacc_data_type = sacc_data_type
+        self.systematics = systematics or []
+        self.data_vector = None
+        self.theory_vector = None
+        self.number_density_func = number_density_func
+        if SupportedTracerNames[sacc_tracer].value == 2:
+            self.mu_p0 = parameters.create()
+            self.mu_p1 = parameters.create()
+            self.mu_p2 = parameters.create()
+            self.sigma_p0 = parameters.create()
+            self.sigma_p1 = parameters.create()
+            self.sigma_p2 = parameters.create()
+        try:
+            self.ccl_kind = SupportedDataTypes[sacc_data_type].name
+        except KeyError:
+            supported_data = [data.name for data in SupportedDataTypes]
+            raise KeyError(
+                f"The SACC data type '{sacc_data_type}' is not "
+                f"supported!"
+                f"Supported names are: {supported_data}"
+            ) from None
+        try:
+            self.tracer_name = SupportedTracerNames[sacc_tracer].name
+        except KeyError:
+            supported_tracers = [tracer.name for tracer in SupportedTracerNames]
+            raise KeyError(
+                f"The SACC tracer name '{sacc_tracer}' is not "
+                f"supported!"
+                f"Supported names are: {supported_tracers}"
+            ) from None
+
+    def _reset(self) -> None:
+        """Reset this systematic.
+
+        This implementation has nothing to do."""
+
+    def _update(self, params: ParamsMap) -> None:
+        """Reset this systematic.
+
+        This implementation has nothing to do."""
+
+    def _required_parameters(self) -> RequiredParameters:
+        return RequiredParameters([])
+
+    @final
+    def _get_derived_parameters(self) -> DerivedParameterCollection:
+        derived_parameters = DerivedParameterCollection([])
+
+        return derived_parameters
+
+    def _compute_grids(self, cosmo, lnN_tuple, logm_tuple, z_tuple, n_intervals=20):
```

Comment:
I fixed the number of 20 because I wanted a reasonable number to perform an MCMC. We still have to decide if this is going to be the truly integration method. If so, we can search for an optimal number or let the user decide. I`m waiting for the NumCosmo conector so maybe I can use a C integration.